gui/serial_unix.cc

```diff
@@ -21,6 +21,68 @@
 #include "mainwindow.h"
 #if !defined (Q_OS_MAC) // FIXME: find a better way to hide this on Mac.
 
+#ifdef HAVE_UDEV
+#include <libudev.h>
+#include <QDebug>
+
+static QStringList dynamicDevices()
+{
+    struct udev *udev = udev_new();
+    if(!udev) {
+        qDebug() << "Can't create udev";
+        return QStringList();
+    }
+
+    QSet<QString> devices;
+
+    struct udev_enumerate *enumerate = udev_enumerate_new(udev);
+    udev_enumerate_add_match_subsystem(enumerate, "tty");
```

Comment:
You know, if we could depend on Qt 5.1 this could be

auto blah = QSerialPortInfo::availablePorts();
for (port : blah) {
   box->addItem(port.portName);
}

Then all our OS-specific versions could be this.

http://doc.qt.io/qt-5/qserialport.html#details


Replacing my Windows/UNIX abstraction of *serial.cc with QSerialPort would
be a little more risky just because that code has been battle tested and
it's really a pain to test any reasonable matrix of OSes and devices.
http://doc.qt.io/qt-5/qserialportinfo.html  It's funny that I wanted a QT
serial system years ago and didn't even noticed that one sneaked in...

At https://www.gpsbabel.org/htmldoc-development/Source.html we set the
floor for Qt at 4.6, which is ridiculously old. Is it time we raise the
floor for Qt?
My read is that our travis instance runs Ubuntu Trusty (2014) and has Qt
5.2 https://lists.ubuntu.com/archives/ubuntu-devel/2014-February/038114.html


On Thu, Mar 31, 2016 at 1:30 PM, Robert Lipe <notifications@github.com>
wrote:

> In gui/serial_unix.cc
> <https://github.com/gpsbabel/gpsbabel/pull/42#discussion_r58104108>:
>
> > +#ifdef HAVE_UDEV
> > +#include <libudev.h>
> > +#include <QDebug>
> > +
> > +static QStringList dynamicDevices()
> > +{
> > +    struct udev *udev = udev_new();
> > +    if(!udev) {
> > +        qDebug() << "Can't create udev";
> > +        return QStringList();
> > +    }
> > +
> > +    QSet<QString> devices;
> > +
> > +    struct udev_enumerate *enumerate = udev_enumerate_new(udev);
> > +    udev_enumerate_add_match_subsystem(enumerate, "tty");
>
> Will this get the bluetooth and USB/serial devices? Seems like it should,
> but it's perhaps an edge case.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/gpsbabel/gpsbabel/pull/42/files/ac44161c6b5f4c4b0e74894850eae59fcffb56fa#r58104108>
>